server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -109,6 +106,7 @@
 {
   private static final EmittingLogger log = new EmittingLogger(RealtimePlumber.class);
   private static final int WARN_DELAY = 1000;
+  private static final int MAX_ROW_EXCEED_CHECK_COUNT = 100000;
```

Comment:
Maybe this is unnecessary/unrealistic, but should this be determined based on the configured maxRowsInMemory? i.e., if the user configures maxRowsInMemory < 100000, they could in theory run into the same OOM issue.

I assume this wouldn't be a practical problem though unless the per-row size is colossal and/or available memory is tiny.